tests/png/bug00295.c

```diff
@@ -0,0 +1,39 @@
+/**
+ * Regression test for <https://github.com/libgd/libgd/issues/295>.
+ *
+ * We're testing that reading a PNG which might trigger a libpng warning
+ * does not trigger a GD_ERROR.
+ */
+
+
+#include "gd.h"
+#include "gd_errors.h"
+#include "gdtest.h"
+#include <string.h>
+
+
+static void error_handler(int priority, const char *format, va_list args)
+{
+    char msg[256];

```

Comment:
imo, if it's in POSIX, it's fine.  if we find a system that lacks the function, we can figure out how to make the function available to it as a fallback.  but better to just wait for someone to complain than always try to make things work for a hypothetical setup that never materializes.

vsnprintf is in POSIX: http://pubs.opengroup.org/onlinepubs/9699919799/functions/vsnprintf.html

it shouldn't be hard to find replacements out there we can drop in (like [gnulib](https://www.gnu.org/software/gnulib/)).